### DISCUSSION TITLE
Bootstrap script to open a PR instead of direct commit

### DISCUSSION THREAD

**Discussion by mixeract** [fluxcd/flux2#4408] [Proposals]
Bootstrap script commit directly to the repo using the github Token provided by the user, however for repositories that have very strict policy (not allowing commits to main directly and require Approved Pull Request) the bootstrap script won't work and we had to implement or re-write the bootstrap script. 

Is there a way for the bootstrap to submit a Pull request with the changes needed, once the PR is merged that's when the cluster will get bootstrapped.

**Comment by stefanprodan** [comment]
Bootstrap is not a script, it is pure Go code with no dependencies to the `git` CLI nor to any OS packages, you can find the code here https://github.com/fluxcd/flux2/tree/main/pkg/bootstrap. Pull requests are a GitHub proprietary API, there is no PR in the Git API. I don't see how we can make bootstrap work with PRs, it would mean for the CLI to block and keep running until the PR is merged, and only then deploy Flux on the cluster, which feels terrible, who wants to keep a CLI running for hours/days.
